### PR TITLE
All Interests 

### PR DESCRIPTION
**Pull request by yosefalnajjarofficial**
- **Wrote the function for all of the interests**
- **Wrote the query** 
   - The query returns all the interests with their id's in the DB 
- **Wrote the tests for both the route and query**

- For running the tests
```
npm test;
```
- A  successful  response 
```js
[
    {
        "id": 1,
        "name": "music"
    },
    {
        "id": 2,
        "name": "sports"
    },
    {
        "id": 3,
        "name": "football"
    }
]
```
#23

### CODE REVIEW

File: test/queries/getInterests.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const test = require('tape');
+const { interests: { getInterests } } = require('../../server/database/queries');
+const dbBuild = require('../../server/database/config/dbbuild');
+
+test('Testing for the interests query', (t) => {
+  dbBuild()
+    .then(() => getInterests())
+    .then((res) => {
+      t.equal(JSON.stringify(res.rows).includes('name'), true, 'There is a name if we have interests');
```
Comment (IsraaSulaiman) [inline test/queries/getInterests.test.js]: Why JSON.strinfigy()! Make a serious test or don't make any tests at all :grimacing: Test, for example, the length of interests array, ot their values.

**Comment by Fatmasiam** [comment]
@yosefalnajjarofficial  put the PR on the project in the next time

File: server/database/queries/interests/getInterests.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ const dbConnection = require('../../config/dbConnection');
 
 exports.getInterests = () => {
   const sql = {
-    text: '',
+    text: 'SELECT * FROM interests',
     values: [],
```
Comment (Fatmasiam) [inline server/database/queries/interests/getInterests.js]:
```suggestion
    
```
if there are no values on your query then remove the values and you can refactor your code 
so you can write it like this 
```js
exports.getInterests = () => {
  const sql = 'SELECT * FROM interests';
  return dbConnection.query(sql);
};

```

File: test/queries/getInterests.test.js (hunk 1 above)
Comment (yosefalnajjarofficial) [reply test/queries/getInterests.test.js]: We don't have a specific number of interests yet so I just made it into a string so I can check if it includes something that I would expect to see in the response

File: server/controllers/interests/getInterest.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
-exports.getAllInterests = (req, res) => {
-  // Here we will get all users with the given interest in the params
-  res.send('Hi from get all interest');
+const { interests: { getInterests } } = require('../../database/queries');
+
+exports.getAllInterests = (req, res, next) => {
+  getInterests()
+    .then(({ rows }) => {
+      res.status(200).json(rows);
```
Comment (Fatmasiam) [inline server/controllers/interests/getInterest.js]:
```suggestion
      res.json({data: rows});
```
```suggestion
      res.status(200).json({data: rows});
```

File: test/queries/getInterests.test.js (hunk 1 above)
Comment (Fatmasiam) [reply test/queries/getInterests.test.js]: thx can review the tests for all PR  @IsraaSulaiman  if you have time for sure

File: .travis.yml [hunk 4]
```diff
@@ -5,4 +5,5 @@ services:
   - postgresql
 
 before_script:
-  - psql -c 'create database travis_ci_test;' -U postgres
+  - psql -c "CREATE DATABASE langmatetest;" -U postgres
```
Comment (Fatmasiam) [inline .travis.yml]: i think you edit this file on another PR so why you are pushing edit also on this PR?!

Comment (Fatmasiam) [inline .travis.yml]: why this file still here?

File: server/controllers/interests/getInterest.js [hunk 5]
```diff
@@ -1,4 +1,7 @@
-exports.getAllInterests = (req, res) => {
-  // Here we will get all users with the given interest in the params
-  res.send('Hi from get all interest');
+const { interests: { getInterests } } = require('../../database/queries');
+
+exports.getAllInterests = (req, res, next) => {
+  getInterests()
+    .then(({ rows }) => res.status(200).json({ data: rows }))
```
Comment (Fatmasiam) [inline server/controllers/interests/getInterest.js]:
```suggestion
    .then(({ rows }) => res.status(200).json({ isSuccess : true , data: rows }))
```

File: server/database/queries/interests/getInterests.js [hunk 6]
```diff
@@ -1,9 +1,6 @@
 const dbConnection = require('../../config/dbConnection');
 
 exports.getInterests = () => {
-  const sql = {
-    text: '',
-    values: [],
-  };
+  const sql = 'SELECT * FROM interests';
```
Comment (11janci) [inline server/database/queries/interests/getInterests.js]: just something to keep in mind, returning all of the records could kill your server if the DB was large enough ;) If you really need to get all of the records, you'd have to implement a paging mechanism to do it safely (but don't bother with that now)...

Comment (Fatmasiam) [reply server/database/queries/interests/getInterests.js]: @11janci, so it's better to specify the record that we need it?
Like 
```js
  SELECT id, name FROM interests
```

Comment (yosefalnajjarofficial) [reply server/database/queries/interests/getInterests.js]: @11janci thanks for the advice, I get what you mean but the interests are not going to be that much even if it's not a fake data

Comment (11janci) [reply server/database/queries/interests/getInterests.js]: @Fatmasiam what I meant is that if the table had 2 million records, this query would select all of them, making the app less responsive, if not killing it completely. But yes as @yosefalnajjarofficial said it should be fine for this table at this scale, just thought I'd point it out as something to keep in mind and watch out for..

**Pull request merged**